dao/src/main/java/greencity/repository/HabitRepo.java

```diff
@@ -70,4 +70,14 @@ public interface HabitRepo extends JpaRepository<Habit, Long>, JpaSpecificationE
         + "WHERE ha.habit_id =:habitId "
         + "AND ha.user_id =:userId", nativeQuery = true)
     List<Long> findHabitAssignByHabitIdAndHabitOwnerId(@Param("habitId") Long habitId, @Param("userId") Long userId);
+
+    /**
+     * Finds a random habit that is not marked as deleted.
+     *
+     * @return a random {@link Habit} entity that is not deleted, or {@code null} if
+     *         no such habit exists.
+     */
+    @Query(nativeQuery = true,
+        value = "SELECT * FROM habits h WHERE h.is_deleted = false ORDER BY random() LIMIT 1")
```

Comment:
This query would not be efficient due to a sorting of the table underhood the ```ORDER BY random()```. You can try using ```TABLESAMPLE SYSTEM()``` to take a small part of the table and then operate with it. Also, we don't need the alias h here because you are selecting from only one table